asdf/tags/core/tests/test_ndarray.py

```diff
@@ -878,6 +878,13 @@ def test_readonly(tmpdir):
             af["data"][0] = 41
             assert str(err) == "assignment destination is read-only"
 
+    # Forcing memmap, the array should still be readonly
+    with asdf.open(tmpfile, copy_arrays=False) as af:
+        assert af["data"].flags.writeable is False
+        with pytest.raises(ValueError) as err:
+            af["data"][0] = 41
+            assert str(err) == "assignment destination is read-only"
```

Comment:
```suggestion
        with pytest.raises(ValueError, match=r"assignment destination is read-only"):
            af["data"][0] = 41
```
The `match` API is a little nicer

Reply:
Thanks! I updated the new test and the one above it that performs the same check with defaults arguments to asdf.open.